sharding/indexHashedNodesCoordinatorWithRater.go

```diff
@@ -54,28 +54,32 @@ func NewIndexHashedNodesCoordinatorWithRater(
 	return ihncr, nil
 }
 
-// ComputeLeaving - computes the validators that have a threshold below the minimum rating
-func (ihgs *indexHashedNodesCoordinatorWithRater) ComputeLeaving(allValidators []*state.ShardValidatorInfo) ([]Validator, error) {
-	leavingList := make([]Validator, 0)
+// ComputeAdditionalLeaving - computes the extra leaving validators that have a threshold below the minimum rating
+func (ihgs *indexHashedNodesCoordinatorWithRater) ComputeAdditionalLeaving(allValidators []*state.ShardValidatorInfo) (map[uint32][]Validator, error) {
+	extraLeavingNodesMap := make(map[uint32][]Validator, 0)
 	minChances := ihgs.GetChance(0)
 	for _, vInfo := range allValidators {
 		if vInfo.List == string(core.InactiveList) {
 			log.Debug("inactive validator", "pk", vInfo.GetPublicKey())
 			continue
 		}
+		if vInfo.List == string(core.LeavingList) {
+			log.Debug("leaving validator from trie", "pk", vInfo.GetPublicKey())
```

Comment:
when shuffled out because of bad rating, does the validator not go into the leaving list ? In case he is there, how is the sorting ?

Reply:
the leavingList from the trie is computed in the computeNodesConfigFromList, with eligible, waiting and new. They are sorted according to the index in both maps.